pkg/build/webhook/generic/generic.go

```diff
@@ -65,8 +66,13 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 		}
 
 		var data api.GenericWebHookEvent
+		body, err = yaml.ToJSON(body)
```

Comment:
yaml.ToJSON no-ops if the content is json. 

so this way we will "work" even if they screw up the content-type header, and no real additional cost.